src/components/VoteCard.tsx

```diff
@@ -1,97 +1,297 @@
+
+	/* 
+	Dynamically create the className for the candidate's reason for voting
+	*/
+	const reasonVoteClassName = `reason-vote vote-${vote.candidateVote.toLowerCase()}`;
+
+	/* 
+	Dynamically calculate the total number of votes
+	*/
+	const totalvotes= `${vote.result.total.yes + vote.result.total.no + vote.result.total.abstain + vote.result.total.none}`
+
+
+
+	/* 
+	Let Typescript know that candidateVote will always have one of these four values. 
+	The cleaner refactored version of this would be to change "vote" into it's own datatype where vote.candidateVote is 
+	predefined to only have one of these four values.
+	*/
+	const voteString: "yes" | "no" | "abstain" | "none" = vote.candidateVote
+
+
+	
+	return (
+		<div className="grey-background">
+			<IonCard className="vote-card" onClick={() => setShowDetails(!showDetails)}>
+				{/* 
+				Card header with Vote name, vote subtitle which holds the type of vote it was and the candidate's vote.
+				It's inside a css grid so we can more easily arrange the items.
+				*/}
+				<IonCardHeader className="vote-card-header">
+					<IonGrid>
+						<IonRow>
+							<IonCol size="8">
+								<IonCardSubtitle className="vote-card-subtitle">
+									{vote.subtitle}
+								</IonCardSubtitle>
+								<IonCardTitle className="vote-card-title">
+									{vote.title}
+								</IonCardTitle>
+							</IonCol>
+							<IonCol size="4">
+								<div className={candidateVoteClassName}>
+									{voteStrings[voteString]}
+								</div>
+							</IonCol>
+						</IonRow>
+					</IonGrid>
+				</IonCardHeader>
+
+				<IonCardContent className="vote-card-content">
+					{/* We also use a grid here to more easily arrange the indivdual components. */}
+					<IonGrid>
+						{/* Vote Result Chart */}
+						<IonRow>
+							<IonCol size="12">
+								<div className="vote-chart-container">
+									{/* Render a VoteChart component for the vote result */}
+									<VoteChart vote={vote} />
+								</div>
+							</IonCol>
+						</IonRow>
+
+						{/* Vote Result Legend */}
+						<IonRow>
+							<IonCol size="1"></IonCol>
+							<IonCol size="2">
+								<div className="vote-legend-circle vote-yes"></div>
+								<span className="vote-legend-text">Ja</span>
+							</IonCol>
+							<IonCol size="2">
+								<div className="vote-legend-circle vote-no"></div>
+								<span className="vote-legend-text">Nein</span>
+							</IonCol>
+							<IonCol size="3">
+								<div className="vote-legend-circle vote-abstain"></div>
+								<span className="vote-legend-text">
+									Enthalten
+								</span>
+							</IonCol>
+							<IonCol size="4">
+								<div className="vote-legend-circle vote-none"></div>
+								<span className="vote-legend-text">
+									Nicht abg.
+								</span>
+							</IonCol>
+						</IonRow>
+					</IonGrid>
+				</IonCardContent>
+			</IonCard>
+
+			{/*Vote Detail Modal*/}
+			
+				<IonContent>

```

Comment:
It might make sense to refactor this into a separate component. Not because it is being reused, but because the file is getting a bit long. But it's ok for now.